src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -3562,6 +3731,137 @@ public boolean updateTypeForSyntheticClasses(
     throw new RuntimeException("Could not find the corresponding missing class!");
   }
 
+  /**
+   * Corrects synthetic method arguments in which a method call contained a method reference.
+   *
+   * @param methodReferencesToCorrect a Map containing the method reference usage (Bar::method) to
+   *     the correct parameter types
+   * @return true if any method argument was updated
+   */
+  @SuppressWarnings("argument")
+  // found: int, required: @Interned int
+  public boolean updateMethodReferenceParameters(Map<String, String> methodReferencesToCorrect) {
+    boolean updated = false;
+    for (String methodReference : methodReferencesToCorrect.keySet()) {
+      if (methodRefUsageToSyntheticMethodDef.containsKey(methodReference)) {
+        Map<UnsolvedMethod, Set<Integer>> unsolvedMethodsToArguments =
+            methodRefUsageToSyntheticMethodDef.get(methodReference);
+        for (UnsolvedMethod method : unsolvedMethodsToArguments.keySet()) {
+          Set<Integer> argumentsToFix = unsolvedMethodsToArguments.get(method);
+
+          String parametersAsString = methodReferencesToCorrect.get(methodReference);
+
+          if (parametersAsString == null) {
+            throw new RuntimeException("Expected corrected parameter types from JavaTypeCorrect");
+          }
+
+          List<String> parameters = new ArrayList<>();
+
+          for (String parameter : getReferenceTypesFromCommaSeparatedString(parametersAsString)) {
+            parameters.add(lookupFQNs(parameter.trim()));
+          }
+
+          // 1st phase: Assume non-void until javac says it is void
+          String fixed =
+              resolveFunctionalInterfaceWithFullyQualifiedParameters(
+                  parameters, false, currentPackage);
+
+          for (int argument : argumentsToFix) {
```

Comment:
This for loop is the reason you're getting a warning from the interning checker. The type of `argument` should be `Integer`, not `int` (primitives can't be stored in iterables, so this is hiding a boxing conversion, which is undesirable).

You can call `getValue()` on the `Integer` if you need the primitive.

Reply:
Ah, that makes sense. I was wondering where that warning was coming from; thanks!